### PR TITLE
store more than largest kube api request amount in the event store

### LINKED ISSUE #1592: Event buffer is too small for big clusters
**Describe the bug**

Lens currently uses `limit=1000` in `EventStore` to protect kubernetes api server. The same `1000` is also used for internal event buffer size which does not make that much of sense. Buffer will go full immediately after event query is done and first event comes from watch API -> Lens starts to loose events.

### PR DESCRIPTION
**Pull request by Nokel81**
Signed-off-by: Sebastian Malton <sebastian@malton.name>

fixes #1592

### CODE REVIEW

File: src/renderer/kube-object.store.ts [hunk 1]
```diff
@@ -11,7 +11,8 @@ import { getHostedCluster } from "../common/cluster-store";
 @autobind()
 export abstract class KubeObjectStore<T extends KubeObject = any> extends ItemStore<T> {
   abstract api: KubeApi<T>;
-  public limit: number;
+  public readonly abstract limit: number;
```
Comment (panuhorsmalahti) [inline src/renderer/kube-object.store.ts]: We could call limit `requestLimit`

**Review by panuhorsmalahti — APPROVED** [review]

File: src/renderer/kube-object.store.ts (hunk 1 above)
Comment (jakolehm) [reply src/renderer/kube-object.store.ts]: Changing name would be a breaking change.. even `abstract` is breaking change (CI shows failures).

File: src/renderer/kube-object.store.ts [hunk 2]
```diff
@@ -11,7 +11,8 @@ import { getHostedCluster } from "../common/cluster-store";
...
+  public readonly abstract saveLimit: number;
```
Comment (jakolehm) [inline src/renderer/kube-object.store.ts]: Same here, probably cannot be abstract. Also `bufferSize` might be a better name?

**Review by jakolehm — CHANGES_REQUESTED** [review]

File: src/renderer/kube-object.store.ts (hunk 1 above)
Comment (Nokel81) [reply src/renderer/kube-object.store.ts]: @jakolehm Removed `abstract` can you review again, please?

**Comment by Nokel81** [comment]
@jakolehm Would you be able to take a look at this again, thanks!

File: src/renderer/kube-object.store.ts [hunk 3]
```diff
@@ -11,14 +11,25 @@ import { getHostedCluster } from "../common/cluster-store";
...
   constructor() {
     super();
     this.bindWatchEventsUpdater();
     kubeWatchApi.addListener(this, this.onWatchApiEvent);
   }
 
+  get query(): IKubeApiQueryParams {
+    const { limit } = this;
+
+    if (!limit) {
```
Comment (ixrock) [inline src/renderer/kube-object.store.ts:26]: So `{limit: 0}` doesn't make any sense?

Comment (Nokel81) [reply src/renderer/kube-object.store.ts:26]: This was just a direct move from the old line 66 below. I don't know... does kube treat `limit: 0` as the same as not passing `limit` at all? (given than kube is written in Go, that is probably the case sadly).

File: src/renderer/kube-object.store.ts [hunk 4]
```diff
@@ -11,14 +11,25 @@ import { getHostedCluster } from "../common/cluster-store";
 @autobind()
 export abstract class KubeObjectStore<T extends KubeObject = any> extends ItemStore<T> {
   abstract api: KubeApi<T>;
-  public limit: number;
+  public readonly limit?: number;
+  public readonly bufferSize: number = 50000;
```
Comment (ixrock) [inline src/renderer/kube-object.store.ts:15]:
```suggestion
  public readonly maxLimit: number = 50000;
```

**Review by ixrock — APPROVED** [review]
LGTM

File: src/renderer/kube-object.store.ts (hunk 4 above)
Comment (ixrock) [reply src/renderer/kube-object.store.ts:15]: or `saveLimit` maybe?

Comment (Nokel81) [reply src/renderer/kube-object.store.ts:15]: I had `saveLimit` initially, Jari suggested `bufferSize`

**Review by jakolehm — APPROVED** [review]

**Pull request merged**